netlogo-gui/src/main/app/Preference.scala

```diff
@@ -0,0 +1,41 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.app
+
+import java.util.Locale
+import java.util.prefs.Preferences
+import javax.swing.{ JComboBox, JComponent }
+
+import org.nlogo.core.I18N
+
+trait Preference {
+  val i18nKey: String
+  val component: JComponent
+  val restartRequired: Boolean
+  def load(prefs: Preferences): Unit
+  def save(prefs: Preferences): Unit
+}
+
+object Preference {
+  object Language extends Preference {
+    case class LocaleWrapper(val locale: Locale) {
```

Comment:
The best name I came up with. If I should change it to something else, let me know